### ISSUE TITLE
Broken Decoding of nil map value in map

### ISSUE THREAD

**Issue by ajwerner** [ugorji/go#208]
https://github.com/ugorji/go/commit/54210f4e076c57f351166f0ed60e67d3fca57a36
Seems to have broken decoding of nil map values. They are now omitted.

The below test demonstrates this behavior. In the prior commit it passes.
```
func TestMsgpackDecodeNilMapValue(t *testing.T) {
        type Struct struct {
                Field map[uint16]map[uint32]struct{}
        }
        toEncode := Struct{Field: map[uint16]map[uint32]struct{}{
                1: nil,
        }}
        handle := &MsgpackHandle{}
        handle.MapType = reflect.TypeOf(map[interface{}]interface{}(nil))
        buf := bytes.NewBuffer(nil)
        enc := NewEncoder(buf, handle)
        if err := enc.Encode(toEncode); err != nil {
                t.Fatalf("Failed to encode struct: %v", err)
        }
        dec := NewDecoder(bytes.NewReader(buf.Bytes()), handle)
        var decoded Struct
        if err := dec.Decode(&decoded); err != nil {
                t.Fatalf("Failed to decode struct: %v", err)
        }
        if !reflect.DeepEqual(decoded, toEncode) {
                t.Fatalf("Decoded value %#v != %#v", decoded, toEncode)
        }
}
```

**Comment by ajwerner** [comment]
https://github.com/ugorji/go/blob/master/codec/codec_test.go#L1753

seems like the relevant testing TODO

**Comment by ugorji** [comment]
Fixed via 50530d8d1478c8bbfbdb18cbf5465db9aee6ada4
